File: csharp/Agentic.Commerce.Protocol/Models/CardNumberType.cs
// this file is @generated
using System.Runtime.Serialization;
using Newtonsoft.Json;
using Newtonsoft.Json.Converters;

namespace Agentic.Commerce.Protocol
{
    [JsonConverter(typeof(StringEnumConverter))]
    public enum CardNumberType
    {
        [EnumMember(Value = "fpan")]
        Fpan = 1,

        [EnumMember(Value = "network_token")]
        NetworkToken = 2,
    }
}

File: kotlin/src/main/kotlin/com/svix/agentic_commerce_protocol/CardNumberType.kt
// This file is @generated
package com.svix.agentic_commerce_protocol

import kotlinx.serialization.SerialName
import kotlinx.serialization.Serializable
import kotlinx.serialization.json.Json
import kotlinx.serialization.json.encodeToJsonElement
import kotlinx.serialization.json.jsonPrimitive

@Serializable
enum class CardNumberType : ToQueryParam {
    @SerialName("fpan") FPAN,
    @SerialName("network_token") NETWORK_TOKEN;

    override fun toQueryParam() = Json.encodeToJsonElement(this).jsonPrimitive.content
}
